src/System.Reflection.DispatchProxy/src/System/Reflection/DispatchProxyGenerator.cs

```diff
@@ -426,13 +426,68 @@ internal void AddInterfaceImpl(Type iface)
                 _assembly.EnsureTypeIsVisible(iface);
 
                 _tb.AddInterfaceImplementation(iface);
+
+                // AccessorMethods -> Metadata mappings.
+                var propertyMap = new Dictionary<MethodInfo, PropertyAccessorInfo>();
```

Comment:
Thank you for suggestion. It sounds good, but determining with `IsSpecialName` is not perfect. Implementing naive `EqualityComparer` as following seems better:
* Compare method's `DeclaringType`
* Compare method's `Name`
* Compare method's `ReturnType`
* Compare method's parameters count.
* Compare each parameters' `ParameterType` of the method.